### PR TITLE
TEST PR DO NOT MERGE - Test CI rework

### PR DESCRIPTION
**Pull request by soenkeliebau**
This is a test PR to test the ongoing rework of ci stuff.

Not intended to be merged.

### CODE REVIEW

File: .github/workflows/all.yml [hunk 1]
```diff
@@ -0,0 +1,280 @@
...
+  check_charts:
+    name: Check if committed Helm & Kustomize Charts were up to date
+    needs:
+      - run_tests
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
```
Comment (lfrancke) [inline .github/workflows/all.yml:198]: this checks out everything, is that intended?
Do we need all the history?

File: .github/workflows/all.yml [hunk 2]
```diff
@@ -0,0 +1,280 @@
...
+    env:
+      NEXUS_PASSWORD: ${{ secrets.NEXUS_PASSWORD }}
+      REPO: ${{ needs.select_repo.outputs.repository }}
+    if: needs.version.outputs.repository != 'skip'
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Package Chart
+        run: mkdir -p target/helm && helm package --destination target/helm deploy/helm/${{ env.PRODUCT_NAME }}-operator
+      - name: Build Docker image
```
Comment (lfrancke) [inline .github/workflows/all.yml:270]: I know this is WIP so if it's on your radar great: Building Docker image &helm charts should happen before the chart tests and the generated image & chart should be passed between jobs and then only published here I believe

File: .github/workflows/all.yml (hunk 1 above)
Comment (soenkeliebau) [reply .github/workflows/all.yml:204]: That was a copy & paste remnant..

File: .github/workflows/all.yml (hunk 2 above)
Comment (soenkeliebau) [reply .github/workflows/all.yml:293]: I wanted to punt on that for now tbh..

**Pull request closed without merging**